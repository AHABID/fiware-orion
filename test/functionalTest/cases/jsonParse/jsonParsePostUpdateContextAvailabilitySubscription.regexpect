1: ***********************************
HTTP/1.1 200 OK
Content-Length: 143
Content-Type: application/json
Date: REGEX(.*)

{
    "errorCode": {
        "code": "404",
        "reasonPhrase": "No context element found"
    },
    "subscriptionId": "012345678901234567890123"
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1M",
    "subscriptionId": "REGEX([0-9a-f]{24})"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: REGEX(.*)

{
    "duration": "PT1S",
    "subscriptionId": "REGEX([0-9a-f]{24})"
}
4: ***********************************
