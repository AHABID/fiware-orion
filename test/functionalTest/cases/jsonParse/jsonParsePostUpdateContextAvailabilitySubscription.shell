echo "1: ***********************************"
url="/NGSI9/updateContextAvailabilitySubscription"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "duration": "PT1M",
  "restriction": {
    "attributeExpression" : "AE",
    "scopes": [
      {
        "type" : "st1",
        "value" : "1"
      },
      {
        "type" : "st2",
        "value" : "2"
      }
    ]
  },
  "subscriptionId" : "012345678901234567890123"
}'
curlJson ${url} "${payload}"

echo "2: ***********************************"

url="/NGSI9/subscribeContextAvailability"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "reference": "http://10.1.1.1:80/test/interfaceNotification",
  "duration": "PT1M",
  "restriction" : {
    "attributeExpression" : "ATTR_EXPR",
    "scopes" : [
      {
        "value" : "sv1",
        "type"  : "st1"
      },
      {
        "value" : "sv2",
        "type"  : "st2"
      }
    ]
  }
}'
curlJson ${url} "${payload}"

SUB_ID=$(echo "$response" | grep subscriptionId | awk -F\" '{ print $4}')

echo "3: ***********************************"

url="/NGSI9/updateContextAvailabilitySubscription"
payload='{
  "entities": [
    {
      "type": "Room",
      "isPattern": "false",
      "id": "ConferenceRoom"
    },
    {
      "type": "Room",
      "isPattern": "false",
      "id": "OfficeRoom"
    }
  ],
  "attributes": [
    "temperature",
    "occupancy",
    "lightstatus"
  ],
  "duration": "PT1S",
  "restriction": {
    "attributeExpression" : "AE",
    "scopes": [
      {
        "type" : "st1",
        "value" : "1"
      },
      {
        "type" : "st2",
        "value" : "2"
      }
    ]
  },
  "subscriptionId" : "'${SUB_ID}'"
}'

curlJson ${url} "${payload}"

echo "4: ***********************************"
