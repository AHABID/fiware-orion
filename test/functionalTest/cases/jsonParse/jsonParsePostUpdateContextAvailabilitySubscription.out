1: ***********************************
HTTP/1.1 200 OK
Content-Length: 143
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:35 GMT

{
    "errorCode": {
        "code": "404", 
        "reasonPhrase": "No context element found"
    }, 
    "subscriptionId": "012345678901234567890123"
}
2: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:35 GMT

{
    "duration": "PT1M", 
    "subscriptionId": "538c340f9844f26710fdded3"
}
3: ***********************************
HTTP/1.1 200 OK
Content-Length: 75
Content-Type: application/json
Date: Mon, 02 Jun 2014 08:21:36 GMT

{
    "duration": "PT1S", 
    "subscriptionId": "538c340f9844f26710fdded3"
}
4: ***********************************
